dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGVisualizer.java

```diff
@@ -48,116 +49,135 @@
      */
-    void visualizeStoreMethodVals(FlowExpressions.MethodCall methodCall, A value);
+    String visualizeStoreMethodVals(FlowExpressions.MethodCall methodCall, A value);
 
     /**
      * Called by {@code CFAbstractStore#internalVisualize()} to visualize the value of class names
-     * collected by this Store.
+     * collected by this store.
      *
      * @param className the class name
      * @param value the value of the class name
+     * @return the String representation of the class names collected by this store
      */
-    void visualizeStoreClassVals(FlowExpressions.ClassName className, A value);
+    String visualizeStoreClassVals(FlowExpressions.ClassName className, A value);
 
     /**
      * Called by {@code CFAbstractStore#internalVisualize()} to visualize the specific information
-     * collected according to the specific kind of Store. Currently, these Stores call this method:
+     * collected according to the specific kind of store. Currently, these Stores call this method:
      * {@code LockStore}, {@code NullnessStore}, and {@code InitializationStore} to visualize
      * additional information.
      *
      * @param keyName the name of the specific information to be visualized
      * @param value the value of the specific information to be visualized
+     * @return the String representation of the specific information according to the specific kind
+     *     of store
      */
-    void visualizeStoreKeyVal(String keyName, Object value);
+    String visualizeStoreKeyVal(String keyName, Object value);
 
     /**
      * Called by {@code CFAbstractStore} to visualize any information after the invocation of {@code
      * CFAbstractStore#internalVisualize()}.
+     *
+     * @return the String representation of the footer to use for a store
      */
-    void visualizeStoreFooter();
+    String visualizeStoreFooter();
 
     /**
-     * Visualize a block based on the analysis.
+     * Visualize a special block based on the analysis.
      *
-     * @param bb the block
+     * @param bb the special block
      * @param analysis the current analysis
+     * @return the String representation of this block
      */
-    void visualizeBlock(Block bb, @Nullable Analysis<A, S, T> analysis);
+    String visualizeBlock(Block bb, @Nullable Analysis<A, S, T> analysis);
 
     /**
-     * Visualize a SpecialBlock.
+     * Visualize a special block.
      *
-     * @param sbb the special block
+     * @param sbb the special block to visualize
+     * @return the String representation of the type of this special block (entry, exit or
+     *     exceptional-exit)
      */
-    void visualizeSpecialBlock(SpecialBlock sbb);
+    String visualizeSpecialBlock(SpecialBlock sbb);
 
     /**
      * Visualize the transferInput of a Block based on the analysis.
      *
      * @param bb the block
      * @param analysis the current analysis
+     * @return the String representation of the transferInput of this block
```

Comment:
Is it "this block" (which implies that the receiver is or has a block), or "the given block" (which then refers to `bb`)?